tests/z_test_fragment_rx.c

```diff
@@ -0,0 +1,90 @@
+//
```

Comment:
Why not to integrate the test in the `client` and/or `multicast` tests instead of having a new one? Also, because this shall be tested while interoperating with the Rust implementation.

Reply:
I agree with both, but the fact we don't have a test that actually test fragmentation with the Rust implementation is the at root of https://github.com/eclipse-zenoh/zenoh-pico/issues/291. So, if I have to prioritise, having a test that tests against Rust is more important than on UDP multicast. Keeping in mind that we don't have reliability on UDP multicast.